### DISCUSSION TITLE
Adding Type Hinting to Searx

### DISCUSSION THREAD

**Discussion by glanham-jr** [searx/searx#3475] [Development]
I was thinking about starting a branch for adding typing hints to searx. I think it would help in grokking the code and reasoning about it.

Would this be something we could merge, or is it generally not worth the effort to maintain after it's implemented?

**Comment by unixfox** [comment]
Highly doubt searx will make any major changes to its code base. If you are interested in an active fork of this project: https://github.com/searxng/searxng

**Reply by glanham-jr** [reply]
It does seem to be pretty inactive - how come it doesn't receive much maintenance anymore? Are the majority of developers now using SearxNG? I self-host so I followed the advise of using this repo.

**Reply by unixfox** [reply]
See https://github.com/searxng/searxng/issues/46

**Reply by virtadpt** [reply]
If you look at the commits in the repo, Searx is in shark mode for the most part.  It's very well suited for what it does and only external factors seem to require any changes.  If engines go offline they'll probably be removed, if new ones are contributed they'll be added if the PR is accepted.

Things don't need to change for the sake of changing just to look really busy.

**Reply by glanham-jr** [reply]
I'm afraid I don't understand the shark mode analogy. But based on what you said and the commits, it seems like this project is considered "complete" - so the priorities are keeping dependencies up to date, fixing engines, and the occasional new feature if it can be merged. I now understand why adding typing is not a priority for this project.

**Reply by virtadpt** [reply]
That is exactly what shark mode means.

**Reply by unixfox** [reply]
> It does seem to be pretty inactive - how come it doesn't receive much maintenance anymore? Are the majority of developers now using SearxNG? I self-host so I followed the advise of using this repo.

Searxng is for everyone, not just for people using public instances. You can easily install searxng: https://docs.searxng.org/admin/installation.html and bugs (especially engines) get fixed way quicker on searxng.

**Reply by glanham-jr** [reply]
The searx readme recommended otherwise - but I think I'll probably switch over then.

**Reply by unixfox** [reply]
I saw that, it's false claims. I submitted changes to the readme: https://github.com/searx/searx/pull/3477
